pow/src/main/java/tech/pegasys/artemis/pow/Eth1DepositManager.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.pow;
+
+import static tech.pegasys.artemis.pow.MinimumGenesisTimeBlockFinder.isBlockAfterMinGenesis;
+import static tech.pegasys.artemis.pow.MinimumGenesisTimeBlockFinder.notifyMinGenesisTimeBlockReached;
+
+import com.google.common.primitives.UnsignedLong;
+import java.math.BigInteger;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.web3j.protocol.core.methods.response.EthBlock;
+import tech.pegasys.artemis.pow.api.Eth1EventsChannel;
+import tech.pegasys.artemis.util.async.AsyncRunner;
+import tech.pegasys.artemis.util.async.SafeFuture;
+import tech.pegasys.artemis.util.config.Constants;
+
+public class Eth1DepositManager {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final Eth1Provider eth1Provider;
+  private final AsyncRunner asyncRunner;
+  private final Eth1EventsChannel eth1EventsChannel;
+  private final DepositProcessingController depositProcessingController;
+  private final MinimumGenesisTimeBlockFinder minimumGenesisTimeBlockFinder;
+
+  public Eth1DepositManager(
+      Eth1Provider eth1Provider,
+      AsyncRunner asyncRunner,
+      Eth1EventsChannel eth1EventsChannel,
+      DepositProcessingController depositProcessingController,
+      MinimumGenesisTimeBlockFinder minimumGenesisTimeBlockFinder) {
+    this.eth1Provider = eth1Provider;
+    this.asyncRunner = asyncRunner;
+    this.eth1EventsChannel = eth1EventsChannel;
+    this.depositProcessingController = depositProcessingController;
+    this.minimumGenesisTimeBlockFinder = minimumGenesisTimeBlockFinder;
+  }
+
+  public void start() {
+    getHead()
+        .thenCompose(
+            headBlock -> {
+              if (isBlockAfterMinGenesis(headBlock)) {
+                return headAfterMinGenesisMode(headBlock);
+              } else {
+                return headBeforeMinGenesisMode(headBlock);
+              }
+            })
+        .finish(
+            () -> LOG.info("Eth1DepositsManager successfully ran startup sequence."),
+            (err) -> LOG.warn("Eth1DepositsManager unable to run startup sequence.", err));
```

Comment:
If we're not able to retry this at all, it probably should be a `fatal` level.  The program is no longer able to function as it should and needs to be restarted.  Definitely at least error level.

Reply:
Done.